File: cpp/test_cases/leetcode_length_of_longest_substring_test.cpp
#include <gtest/gtest.h>

#include "length_of_longest_substring.hpp"

namespace {
auto solution = Solution();

TEST(LeetcodeLengthOfLongestSubstringTest, Test001) {
  EXPECT_EQ(solution.lengthOfLongestSubstring("abcabcbb"), 3);
}

TEST(LeetcodeLengthOfLongestSubstringTest, Test002) {
  EXPECT_EQ(solution.lengthOfLongestSubstring("bbbbb"), 1);
}

TEST(LeetcodeLengthOfLongestSubstringTest, Test003) {
  EXPECT_EQ(solution.lengthOfLongestSubstring("pwwkew"), 3);
}
}  // namespace

File: python/test_cases/leetcode_length_of_longest_substring_test.py
import unittest
from leetcode.length_of_longest_substring import Solution


class LeetcodeLengthOfLongestSubstringTest(unittest.TestCase):
    solution = Solution()

    def test_001(self):
        self.assertEqual(self.solution.lengthOfLongestSubstring(s="abcabcbb"), 3)

    def test_002(self):
        self.assertEqual(self.solution.lengthOfLongestSubstring(s="bbbbb"), 1)

    def test_003(self):
        self.assertEqual(self.solution.lengthOfLongestSubstring(s="pwwkew"), 3)
